analyzer/codechecker_analyzer/cmd/checkers.py

```diff
@@ -255,15 +252,37 @@ def format_guideline(guideline):
         return ' '.join('Related {} rules: {}'.format(g, ', '.join(r))
                         for g, r in guideline.items())
 
+    def guideline_rules_for_checker(checker):
+        """
+        Returns the guideline and rules covered by the given checker. This
+        function returns a dict which maps guideline names to the covered
+        rules: { "sei-cert": ["arr30-c", ...], ... }.
+        """
+        result = defaultdict(list)
+        labels = context.checker_labels.labels_of_checker(checker)
+        guidelines = context.checker_labels.get_description('guideline')
+
+        for label in labels:
+            if label[0] in guidelines:
+                result[label[0]].append(label[1])
+
+        return result
+
+    cl = context.checker_labels
+
+    # TODO: --profile is a deprecated flag. This section should be removed in
+    # the next release.
```

Comment:
This comment refers to the code lines under it, not the `--profile` flag. I would leave this TODO message here, because removal of a deprecated flag doesn't trigger the modification of this part.